src/Tizen.Applications.ComponentBased/Tizen.Applications.ComponentBased.Common/ComponentBasedApplication.cs

```diff
@@ -0,0 +1,201 @@
+            }
+            argsClone[0] = string.Empty;
+
+            Interop.CBApplication.ErrorCode err = Interop.CBApplication.BaseMain(argsClone.Length, argsClone, ref _callbacks, IntPtr.Zero);
+            if (err != Interop.CBApplication.ErrorCode.None)
+            {
+                Log.Error(LogTag, "Failed to run the application. Err = " + err);
+                throw new InvalidOperationException("Fail to run application : err(" + err + ")");
+            }
+        }
+
+        /// <summary>
+        /// Exits the main loop of the application.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public override void Exit()
+        {
+            Interop.CBApplication.BaseExit();
+        }
+
+        private IntPtr OnCreateNative(IntPtr data)
+        {
+            Log.Debug(LogTag, "On create");
+            IntPtr nativeComponentFactoryMap = IntPtr.Zero;
+            foreach (KeyValuePair<Type, ComponentStateManger> entry in _componentFactories)
+            {
+                nativeComponentFactoryMap = entry.Value.Bind(nativeComponentFactoryMap);
+            }
+
+            return nativeComponentFactoryMap;
+        }
+
+        private void OnTerminateNative(IntPtr data)
+        {
+        }
+
+        private void OnRunNative(IntPtr data)
+        {
+            OnRun();
+        }
+
+        private void OnExitNative(IntPtr data)
+        {
+            OnExit();
+        }
+
+        private void OnInitNative(int argc, string[] argv, IntPtr userData)
+        {
+            OnInit(argv);
+        }
+
+        private void OnFinishedNative(IntPtr data)
+        {
+            OnFinished();
+        }
+
+        /// <summary>
+        /// This method will be called before running main-loop
+        /// </summary>
+        /// <param name="args"></param>
+        /// <since_tizen> 6 </since_tizen>
+        protected virtual void OnInit(string[] args)
+        {
+        }
+
+        /// <summary>
+        /// This method will be called after exiting main-loop
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        protected virtual void OnFinished()
+        {
+        }
+
+        /// <summary>
+        /// This method will be called to start main-loop
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        protected abstract void OnRun();
+
+        /// <summary>
+        /// This method will be called to exit main-loop
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        protected virtual void OnExit()
+        {
+        }
+
+        /// <summary>
+        /// Releases any unmanaged resources used by this object. Can also dispose any other disposable objects.
+        /// </summary>
+        /// <param name="disposing">If true, disposes any disposable objects. If false, does not dispose disposable objects.</param>
+        /// <since_tizen> 6 </since_tizen>
+        protected override void Dispose(bool disposing)
```

Comment:
It seem no need to override. 
becuase _disposedValue does not used

Reply:
Removed